### PR TITLE
Missions Revamp

### PR DESCRIPTION
**Pull request by OcelotWalrus**
This PR updates the mission menu, by splitting missions in 3 categories:

**Introduction**:
- Ice River
- Itrium

**Yuruki Campaign**:
- Fillum Resources Claim (_new_)
- Fillum Crater (_new_)

**Synapol Campaign**:
- Crater Defense
- Harbor Defense (_new_)
- Asteroid Invasion (_new_)

**Misc**:
- Firewatch

### CODE REVIEW

**Comment by OcelotWalrus** [comment]
I think it's ready for review now.

**Comment by Mailaender** [comment]
Please submit only one mission per pull request.

File: mods/hv/maps/magma-breach-claim/magma-breach-claim.lua [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+		Human.MarkCompletedObjective(EnemyEliminatedObjective)
+	end
+
+    if DateTime.GameTime % DateTime.Seconds(150) == 0 and DateTime.GameTime > DateTime.Seconds(150) and not Human.IsObjectiveCompleted(EnemyEliminatedObjective) then  -- happens every 2.5 mins, starting at 2.5 game minutes, if the player hasn't killed all units
+        SynapolReinforcements1 = Reinforcements.Reinforce(Enemy, ReinforcementUnits, { SpawningWaypoint1.Location, DestinationWaypoint1.Location })
+        SynapolReinforcements2 = Reinforcements.Reinforce(Enemy, ReinforcementUnits, { SpawningWaypoint2.Location, DestinationWaypoint2.Location })
+        Media.DisplayMessage(UserInterface.Translate("reinforcements-incoming"), Warning)
+    end
+
+    if not Human.IsObjectiveCompleted(ResourcesClaimedObjective) and #towers == 17 then -- if the player has built every mining tower
```
[the remark this answers is not in the document]
Comment (OcelotWalrus) [reply mods/hv/maps/magma-breach-claim/magma-breach-claim.lua:27]: There are 17.

[the remark this answers is not in the document]
Comment (OcelotWalrus) [reply mods/hv/maps/magma-breach-claim/magma-breach-claim.lua:27]: You missed two then:
Screenshot from 2024-05-22 16-22-01

[the remark this answers is not in the document]
Comment (Mailaender) [reply mods/hv/maps/magma-breach-claim/magma-breach-claim.lua:27]: I hope this is not a bug where resources vanish after being depleted.

[the remark this answers is not in the document]
Comment (OcelotWalrus) [reply mods/hv/maps/magma-breach-claim/magma-breach-claim.lua:27]: I think it's just you. I playtested it a lot. When I destroy all synapol structures and started placing every mining towers, they were all at the lowest level.

**Pull request closed without merging**